bundles/org.openhab.binding.boschshc/src/test/java/org/openhab/binding/boschshc/internal/console/BoschShcCommandExtensionTest.java

```diff
@@ -0,0 +1,225 @@
+    }
+
+    @Test
+    void printDeviceInfo() throws InterruptedException {
+        // no bridge
+        when(thingRegistry.getAll()).thenReturn(Collections.emptyList());
+        assertThat(fixture.buildDeviceInfo(), is(""));
+
+        // One bridge, No device
+        BridgeHandler mockBridgeHandler = mock(BridgeHandler.class);
+        Thing mockBridgeThing = mock(Thing.class);
+        when(mockBridgeThing.getLabel()).thenReturn("TestLabel");
+        when(mockBridgeThing.getHandler()).thenReturn(mockBridgeHandler);
+        when(thingRegistry.getAll()).thenReturn(Collections.singletonList(mockBridgeThing));
+        assertThat(fixture.buildDeviceInfo(), allOf(containsString("thing: TestLabel"), containsString("devices (0)")));
+
+        // One bridge, One UNsupported device
+        Device mockShcDevice = mock(Device.class);
+        mockShcDevice.deviceModel = "";
+        mockShcDevice.deviceServiceIds = Collections.emptyList();
+        when(mockBridgeHandler.getDevices()).thenReturn(List.of(mockShcDevice));
+        assertThat(fixture.buildDeviceInfo(), allOf(containsString("thing: TestLabel"), containsString("devices (1)"),
+                containsString("!UNSUPPORTED!")));
+
+        // One bridge, One supported device
+        mockShcDevice.deviceModel = "TWINGUARD";
+        mockShcDevice.deviceServiceIds = Collections.emptyList();
+        when(mockBridgeHandler.getDevices()).thenReturn(List.of(mockShcDevice));
+        assertThat(fixture.buildDeviceInfo(), allOf(containsString("thing: TestLabel"), containsString("devices (1)"),
+                containsString("TWINGUARD -> twinguard")));
+
+        // One bridge, One supported device with services
+        mockShcDevice.deviceModel = "TWINGUARD";
+        mockShcDevice.deviceServiceIds = List.of("unknownService", "batterylevel");
+        when(mockBridgeHandler.getDevices()).thenReturn(List.of(mockShcDevice));
+        assertThat(fixture.buildDeviceInfo(), allOf(containsString("thing: TestLabel"), containsString("devices (1)"),
+                containsString("TWINGUARD -> twinguard"), containsString("service: unknownService -> !UNSUPPORTED!"),
+                containsString("batterylevel -> batterylevel")));
+    }
+
+    @Test
+    void printBindingInfo() {
+        assertThat(fixture.buildBindingInfo(), containsString("Bosch SHC Binding"));
+    }
+
+    @Test
+    void printSupportedDevices() {
+        assertThat(fixture.buildSupportedDeviceStatus(),
+                allOf(containsString("Support Devices"), containsString("BBL = boschshc:shutter-control")));
+    }
+
+    @Test
+    void printSupportedServices() {
+        assertThat(fixture.buildSupportedServiceStatus(),
+                allOf(containsString("Supported Services"), containsString("airqualitylevel")));
+    }
+
+    @Test
+    void getAllBoschShcServices() throws IOException {
+        // The list of services shall match the implemented services
```

Comment:
The single line comments could be turned into a proper Javadoc comment for the method.